### ISSUE TITLE
PR#127 not handling $ref for non-204 statutes

### ISSUE THREAD

**Issue by daareiza** [IBM/openapi-validator#135]
Seems like #127 is not handling correctly responses that references a response in the components node, so it is throwing invalid warnings.

Example:

```yml
paths:
  /authenticate:
    post:
      operationId: create_user_session
      tags:
        - Authentication
      summary: Creates a new user session
      description: Creates a new user session
      security:
        - JWT: []
      responses:
        '200':
          $ref: "#/components/responses/test"
components:
  responses:
    test:
      description: Just a test
      content:
        'application/json': ...
```

**Comment by dpopp07** [comment]
@daareiza Good catch, thanks for opening the issue. We will get on this

**Comment by dpopp07** [comment]
Resolved in #136

**Comment by daareiza** [comment]
@dpopp07 @barrett-schonefeld thanks for the quick fix!!!
